Adding associated pod list screen for service details

rakol-ms opened the pull request:
Signed-off-by: Rajendra Kolli <rakol@microsoft.com>

![svc_details](https://user-images.githubusercontent.com/45386516/50153000-230aef00-02eb-11e9-8d63-f301677beb8c.PNG)

madhuv-msft commented on src/WebUI/Components/ServiceComponent.scss:
```diff
@@ -10,4 +10,24 @@
+
+.s-pod-col-header {
+    color: $secondary-text;
+    font-size: $fontSizeM;
```
do we need this
font-size: $fontSizeM;

madhuv-msft commented on src/WebUI/Components/ServiceComponent.scss:
```diff
@@ -10,4 +10,24 @@
\ No newline at end of file
+}
+
+.s-pod-list-content {
```
.s-pod-list-content {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    margin-top: 10px; 
}

.s-pod-col-header {
    color: $secondary-text;


if these are common settings we should set them in common place

madhuv-msft commented on src/WebUI/Components/ServiceComponent.tsx:
```diff
@@ -14,17 +14,44 @@ import * as Resources from "../Resources";
             <div className="service-main-content">
                 {this._getMainHeading()}
                 {this._getServiceDetails()}
+                {!!this.state.showAssociatedPods && this._getAssociatedPods()}
```
!!this.state.showAssociatedPods && --> we do not need this, as we should show column names when data is not available

madhuv-msft commented on src/WebUI/Components/ServiceComponent.tsx:
```diff
@@ -124,4 +151,117 @@ export class ServiceComponent extends BaseComponent<IServiceComponentProperties>
+    public componentDidMount(): void {
+        console.log("getting items");
+        this.props.podListingPromise.then(podList => {
+            podList.items &&
```
podList && podList.items &&

madhuv-msft commented on src/WebUI/Components/ServiceComponent.tsx:
```diff
@@ -124,4 +151,117 @@ export class ServiceComponent extends BaseComponent<IServiceComponentProperties>
+    }
+
+    private _getAssociatedPods(): JSX.Element | null {
+        if(this.state.showAssociatedPods) {
```
this check is not needed
if(this.state.showAssociatedPods) {

madhuv-msft commented on src/WebUI/Utils.ts:
```diff
@@ -25,4 +26,24 @@ export class Utils {
+        console.log(labels);
+        let labelSelector: string = "";
+        if(labels) {
+            let keySet = Object.keys(labels);
```
const keySet

madhuv-msft commented on src/WebUI/Utils.ts:
```diff
@@ -25,4 +26,24 @@ export class Utils {
+            let keySet = Object.keys(labels);
+            keySet.forEach((key,index) => {
+                labelSelector = labelSelector.concat(format("{0}={1}", key, labels[key]))
+                if (index < keySet.length-1) labelSelector = labelSelector.concat(",");
```
if (index < keySet.length-1), not needed i guess

madhuv-msft commented on src/WebUI/Utils.ts:
```diff
@@ -25,4 +26,24 @@ export class Utils {
+    }
+
+    public static generatePodStatusProps(status:V1PodStatus): IStatusProps {
+        if(status.phase === "Running" || status.phase === "Scheduled"){
```
can we use contracts of the k8s contracts? instead of "Running" etc...

rakol-ms commented:
@madhuv-msft Addressed review comments. PFA the screens for no pods and a svc with pods

![empty_svc_details](https://user-images.githubusercontent.com/45386516/50215248-6e370780-03a8-11e9-8fd7-1a570cfe8d91.PNG)
![svc_details](https://user-images.githubusercontent.com/45386516/50215253-6f683480-03a8-11e9-9e70-12ce76d0737c.PNG)

rakol-ms commented:
![orphanpod](https://user-images.githubusercontent.com/45386516/50266983-bc511700-044b-11e9-9ad4-f7d481517239.PNG)

madhuv-msft commented:
On the latest UI:
we should remove "Orphan Pods" heading. in the same window, make "Pods" as "Pod"
In orphan replicaset window, remove heading of the replicaset and labels for the same. we can talk to PM to show labels as column. we can take that little later.
in this workloads page, "Pods" column should be renamed to "Pods running", this is for deploypment, replicaset, daemon, stateful set tables.

madhuv-msft commented on src/Contracts/Contracts.ts:
```diff
@@ -6,7 +6,7 @@
 
 export interface IKubeService {
-    getPods(): Promise<K8sTypes.V1PodList>;
+    getPods(labelSelector?:string): Promise<K8sTypes.V1PodList>;
```
see if you can add a test for this?

madhuv-msft commented on src/WebUI/Components/KubeSummary.tsx:
```diff
@@ -175,6 +185,7 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
                     key={format("dc-{0}", this.state.namespace || "")}
                     onItemInvoked={this._onDeploymentItemInvoked}
                 />
+                {this.state.showOrphanPods && this.getOrphanPods()}
```
should we move this deploymentscomponent, or rename deploymentscomponent to workloadscomponent?

madhuv-msft commented on src/WebUI/Utils.ts:
```diff
@@ -3,10 +3,22 @@
+/**
+ * https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase
+ */
+export enum PodPhase {
```
if ant used any where we can make this as without export, if used, we can keep this in types.ts file

madhuv-msft commented on tests/Contracts/KubeServiceBase.test.ts:
```diff
@@ -41,4 +41,12 @@ describe("KubeServiceBase Tests", () => {
+    it("getPods calls with labelSelector as input",() => {
+        expect.assertions(1);
+        const labelSelector:string = "app=app";
+        return service.getPods(labelSelector).then(output =>{
```
we can add a case for labelSelector also by maksing changes to MockKubeService class

madhuv-msft commented on src/WebUI/Utils.ts:
```diff
@@ -25,4 +37,24 @@ export class Utils {
+                if (index < keySet.length-1) labelSelector = labelSelector.concat(",");
+            });
+        }
+        return labelSelector;
```
if we want a comma separated list, we can put the labels in to a string and finally can concat
